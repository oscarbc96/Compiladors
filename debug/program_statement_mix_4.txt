FLEX: Detected mode PRGM
BISON: Starting in mode PRGM
FLEX: ASSIGN yytex: :=
FLEX: INT yytex: 3 -> 3
FLEX: IF yytex: if
BISON: Checking mode
FLEX: PARENTHESIS_OPEN yytex: (
FLEX: INT yytex: 3 -> 3
FLEX: COMP yytex: < -> <
FLEX: ID yytex: x -> x
BISON: Looking for ID x
FLEX: OR yytex: or
FLEX: INT yytex: 4 -> 4
FLEX: COMP yytex: = -> =
FLEX: ID yytex: x -> x
BISON: Looking for ID x
FLEX: AND yytex: and
FLEX: NOT yytex: not
FLEX: PARENTHESIS_OPEN yytex: (
FLEX: INT yytex: 3 -> 3
FLEX: COMP yytex: > -> >
FLEX: ID yytex: x -> x
BISON: Looking for ID x
FLEX: OR yytex: or
FLEX: ID yytex: x -> x
BISON: Looking for ID x
FLEX: COMP yytex: = -> =
FLEX: ID yytex: x -> x
BISON: Looking for ID x
FLEX: PARENTHESIS_CLOSE yytex: )
FLEX: PARENTHESIS_CLOSE yytex: )
FLEX: THEN yytex: then
FLEX: ID yytex: x -> x
BISON: Looking for ID x
FLEX: ELSE yytex: else
FLEX: ID yytex: x -> x
FLEX: ASSIGN yytex: :=
FLEX: INT yytex: 1 -> 1
FLEX: FI yytex: fi
The compilation has been successful
