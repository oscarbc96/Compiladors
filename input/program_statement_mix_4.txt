calc off

x:=3

if (3<x or 4=x and not(3>x or x=x)) then
  x
else
  x := 1
fi
