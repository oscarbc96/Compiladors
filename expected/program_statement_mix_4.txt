1: $t01 := 3
2: x := $t01
3: IF 3 LTI x GOTO 11
4: GOTO 5
5: IF 4 EQ x GOTO 7
6: GOTO 14
7: IF 3 GTI x GOTO 14
8: GOTO 9
9: IF x EQ x GOTO 14
10: GOTO 11
11: PARAM x
12: CALL PUTI,1
13: GOTO 16
14: $t02 := 1
15: x := $t02
16: HALT
